latch/types/directory.py

```diff
@@ -114,8 +114,10 @@ def __init__(
         self._path_generated = False
 
         if is_valid_url(self.path) and remote_path is None:
+            self._raw_remote_path = str(path)
             self._remote_directory = self.path
         else:
+            self._raw_remote_path = str(remote_path)
```

Comment:
this can be `"None"` too - likely will never come up unless someone is very confused but may wanna check for it